(Chore) [RN] 1199 Re-integrate hot reloader for web

mauriguz opened the pull request:
# Description

Hot reloading for web was not working on react-native branch, this PR fixes that.

About #1199 


# How Has This Been Tested?

1. Run web and test signup router by making changes in code and let hot reloading do it's job
2. Run web and test logged in router by making changes in code and let hot reloading do it's job
3. Run native and no differences should be found

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

serdiukov-o-nordwhale commented on src/App.js:
```diff
@@ -109,4 +110,4 @@ const AppHolder = () => {
 }
 
-export default AppHolder
+export default hot(AppHolder)
```
why we're wrapping here always but routers are wrapped web only ?

mauriguz replied:
wrapping should only happen for web. fixed.

serdiukov-o-nordwhale commented on src/routers/Router/index.native.js:
```diff
@@ -0,0 +1 @@
+export { default } from './Router'
```
could we keep routes as they are now (directly in src) just 
a) wrap them in SplashOrRouter (if it's needed) 
b) or wrap before export default if it's needed
?

e.g.
```javascript
// SplashOrRouter.js
import Router from './Router'
import SingupRouter from './SingupRouter'

const [HotRouter, HotSingupRouter] = [Router, SingupRouter]
  .map(router => Platform.select({
    web: hot(router),
    default: router
  })
)
```
or
```javascript
// Router.js
// ...
const Router = (//...
// ...

export default Platform.select({
  web: hot(Router),
  default: Router,
})
```

mauriguz replied:
centralised this logic in a utils file to be re-used by any component that needs hot-reloading for web only. fixed.

MarianoAmado commented on src/RouterSelector.web.js:
```diff
@@ -83,7 +83,7 @@ const handleLinks = async () => {
 let SignupRouter = React.lazy(async () => {
   const [module] = await Promise.all([
-    retryImport(() => import(/* webpackChunkName: "signuprouter" */ './SignupRouter')),
+    retryImport(() => import(/* webpackChunkName: "signuprouter" */ './routers/SignupRouter')),
```
Should we change these imports in RouterSelector.native.js too?

mauriguz replied:
with latest changes I reverted changes in routes. fixed.

mauriguz commented:
@serdiukov-o-nordwhale @MarianoAmado I updated the PR with your suggestions, please re-check. I've made the following changes:
1. created util file to wrap components with `hot` depending on the platform
2. reverted changes of routers back to normal
3. fixed iOS errors caused by import of `hot` func
4. added check to never wrap with `hot` when running unit tests - this because it could give false results and break jest

serdiukov-o-nordwhale reviewed (approved):
@mauriguz  I adjusted hotReloadComponent, please check my changes

sirpy commented:
@mauriguz why does it require to wrap both the AppHolder and the two routers? the routers are children of AppHolder
please also fix conflicts and then merge

mauriguz commented:
> @mauriguz I adjusted hotReloadComponent, please check my changes

got it, I'll check. thanks!

> @mauriguz why does it require to wrap both the AppHolder and the two routers? the routers are children of AppHolder
please also fix conflicts and then merge

the wrapping of routers is because they're on different chunks lazy loaded

mauriguz commented on src/lib/hoc/hotLoader.js:
```diff
@@ -0,0 +1,25 @@
+
+if (shouldWrap) {
+  /*
+    hot needs to imported from /root to use their new and more stable API, old API could cause problems
```
```suggestion
    hot needs to be imported from /root to use their new and more stable API, old API could cause problems
```
